packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.tsx

```diff
@@ -0,0 +1,581 @@
+import * as React from 'react';
+import {
+  autobind,
+  BaseComponent,
+  css,
+  findIndex,
+  getId
+} from '../../Utilities';
+import {
+  ISwatchColorPickerProps,
+  ISwatchColorPickerItemProps,
+  SwatchColorPickerItemType
+} from './SwatchColorPicker.Props';
+import { DirectionalHint } from '../../ContextualMenu';
+import { getColorFromString } from '../../utilities/color/colors';
+import { Grid } from '../../utilities/Grid/Grid';
+import { DefaultButton, CommandButton } from '../../Button';
+import { Callout } from '../../Callout';
+import { FocusZone } from '../../FocusZone';
+import * as stylesImport from './SwatchColorPicker.scss';
+const styles: any = stylesImport;
+
+export interface ISwatchColorPickerState {
+  selectedIndex?: number;
+  expanded?: boolean;
+}
+
+export class SwatchColorPicker extends BaseComponent<ISwatchColorPickerProps, ISwatchColorPickerState> {
+
+  public static defaultProps = {
+    cellShape: 'circle',
+    updateButtonIconWithColor: false,
+    selectedId: null,
+    disabled: false
+  };
+
+  private _id: string;
+
+  private _buttonWrapper: HTMLDivElement;
+
+  private _numOfItemsInChunk;
+
+  constructor(props: ISwatchColorPickerProps) {
+    super(props);
+
+    this._id = props.id || getId('swatchColorPicker');
+    this._numOfItemsInChunk = -1;
+
+    this.state = {
+      selectedIndex: props.selectedId && this._getSelectedIndex(props.swatchColorPickerItems, props.selectedId),
+      expanded: false
+    };
+  }
+
+  public componentWillReceiveProps(newProps: ISwatchColorPickerProps) {
+    let newSelectedIndex = newProps.selectedId && this._getSelectedIndex(newProps.swatchColorPickerItems, newProps.selectedId);
+
+    if (newSelectedIndex !== undefined &&
+      newSelectedIndex !== null &&
+      newSelectedIndex !== this.state.selectedIndex) {
+      this.setState({
+        selectedIndex: newSelectedIndex
+      });
+    }
+  }
+
+  public render() {
+    let { swatchColorPickerButtonProps } = this.props;
+
+    // If we got button props, put the swatch color picker behind a button, otherwise
+    // render all of the items
+    let renderElement = swatchColorPickerButtonProps !== undefined ? this._buttonToRender() : this._fullSwatchColorPickerToRender();
```

Comment:
you can pass whatever state you need from the parent component to the other through props. Note that I'm not saying we should export this new component, it should be "private" to this module.